reference/conanfile/attributes.rst

```diff
@@ -458,48 +504,38 @@ consistent implementation take into account these considerations:
 - A different behavior has ``self.options.option = None``, because
   ``assert self.options.option != None``.
 
-If you want to do a safe check of options values, you could use the ``get_safe()`` method:
-
-.. code-block:: python
-
-    def build(self):
-        # Will be None if doesn't exist
-        fpic = self.options.get_safe("fPIC")
-        # Will be the default version if the return is None
-        shared = self.options.get_safe("shared", default=False)
-
-The ``get_safe()`` method will return ``None`` if that option doesn't exist and there is no default value assigned.
 
 
 .. _conanfile_default_options:
 
 default_options
 ---------------
 
-As you have seen in the examples above, recipe's default options are declared as a dictionary with the initial desired value of the options.
-However, you can also specify default option values of the required dependencies:
-
-.. code-block:: python
-
-    class OtherPkg(ConanFile):
-        requires = "pkg/0.1@user/channel"
-        default_options = {"pkg:pkg_option": "value"}
-
-And it also works with default option values of conditional required dependencies:
+The attribute ``default_options`` has the purpose of defining the default values for the options
+if the consumer (consuming recipe, project, profile or the user through the command line) does
+not define them. This attribute should be defined as a python dictionary:
 
 .. code-block:: python
 
-    class OtherPkg(ConanFile):
-        default_options = {"pkg:pkg_option": "value"}
+    class MyPkg(ConanFile):
+        ...
+        options = {"shared": [True, False],
+                   "option1": ["value1", "value2"],
+                   "option2": "ANY"}
+        default_options = {"shared": True,
+                           "option1": "value1",
+                           "option2": 42}
 
-        def requirements(self):
-            if self.settings.os != "Windows":
-                self.requires("pkg/0.1@user/channel")
+        def build(self):
+            shared = "-DBUILD_SHARED_LIBS=ON" if self.options.shared else "-DBUILD_SHARED_LIBS=OFF"
+            cmake = CMake(self)
+            self.run("cmake . %s %s" % (cmake.command_line, shared))
+            ...
 
-For this example running in Windows, the `default_options` for the `pkg/0.1@user/channel` will be ignored, they will only be used on every
-other OS.
+Remember that you can also assign default values for options of your requirements as we've seen in
+the attribute :ref:`conanfile_options`.
 
-You can also set the options conditionally to a final value with ``config_options()`` instead of using ``default_options``:
+You can also set the options conditionally to a final value with ``configure()`` instead of using ``default_options``:
```

Comment:
It seems that the intent of this has been changed. If you change from config_options => configure you are hardcoding the options values. The existing example with config_options was defining a conditional default value, that could be overriden from downstream.

Reply:
From this sentence in the docs _"`config_options()` is used to configure or constraint the available options in a package, before they are given a value."_, people understand that assigning a value in `config_options()` is forbidden (https://github.com/conan-io/conan/issues/7786#issuecomment-702945471), that the method is intended only to restrict the possible values of an option or to remove the option.

Reply:
But it also seems that the CCI relies on the current behavior of `config_options` to allow overridable from downstream default_options: https://github.com/conan-io/conan-center-index/pull/3341#issuecomment-717958354
What's the intended behaviour in here?

Reply:
ATM I think we should describe the current behavior. 

In this section I prefer to use ``configure()`` over ``config_options()`` because it is talking about _"assigning a final value"_ and I understand it means a value that cannot be overridden.